### PR TITLE
Cleanup use of empty strings

### PR DESCRIPTION
**Pull request by tautschnig**
Please review commit-by-commit. The common theme to this PR is a review of all uses of `""`.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 8add28f89d122c871bb10d81f07c22b697261a5a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/99533608

**Review by smowton — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 465e0cce2e77560124ca242b0eda5819838dd40b).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/99770135

**Review by peterschrammel — APPROVED** [review]

**Comment by owen-mc-diffblue** [comment]
The test failure appears to be clang-format complaining about one line

**Review by romainbrenguier — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 68f340094fa9b409455250a044a0c66fdb60e874).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/101054047

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 42fabc16e8bda6126d4d9bf11584133284a016db).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/102434431

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: 233d35080d347f72ed22fa342000d1876a9e9ee3).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/104498814
Status will be re-evaluated on next push.
Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

  - the compatibility was already broken by an earlier merge.

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: 385b7f8b7e89843bae5bb1988cd6da17e72453ba).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/105262938
Status will be re-evaluated on next push.
Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

  - the compatibility was already broken by an earlier merge.

**Review by martin-cs — APPROVED** [review]
I believe this patch does what it claims to.  I think it may even improve performance very slightly but I have to know -- why did you decide to do this?

**Comment by tautschnig** [comment]
> why did you decide to do this?

In some prior PR I had something like `symbol_exprt("", ...)` and someone reminded me that using `irep_idt()` would be more efficient. Fixing that one instance was easy, but I don't really like doing half a job...

File: src/goto-analyzer/taint_analysis.h [hunk 1]
```diff
@@ -22,6 +22,6 @@ bool taint_analysis(
   const std::string &taint_file_name,
   message_handlert &,
   bool show_full,
-  const std::string &json_file_name);
+  const std::string &json_output_file_name = "");
```
Comment (kroening) [inline src/goto-analyzer/taint_analysis.h:26]: How about actually using `optionalt`?

**Comment by kroening** [comment]
On Use s.empty() instead of s == "":
This may be a case of premature optimisation. Note that the implementation of == on std::string and char * is as follows:
```
    size_t __rhs_len = _Traits::length(__rhs);
    if (__rhs_len != __lhs.size()) return false;
    return __lhs.compare(0, _String::npos, __rhs, __rhs_len) == 0;
```
_Traits::length("") will be a compile-time zero; so there is likely no performance benefit. We would need to be convinced that `.empty()` is indeed better readable than `!=""`

**Comment by kroening** [comment]
There are a couple of cases where one might obtain better clarity by using `optionalt` instead of the empty-string as indicator of "no value", but that's beyond this PR.

**Review by kroening — APPROVED** [review]

**Pull request merged**